### PR TITLE
[OTEL-1153] Remap service.name to service in OTel logs

### PR DESCRIPTION
**Pull request by songy23**
https://datadoghq.atlassian.net/browse/OTEL-1153

### What does this PR do? What is the motivation?
Currently there’s a bug with Datadog log exporter that `service` is overridden by `service.name` while `service.name` is not recognized by log preprocessing by default. In order to properly populate service in logs, customers need a log remapper as suggested by https://docs.datadoghq.com/logs/log_configuration/pipelines/?tab=service#service-attribute. Add this caveat to Datadog exporter page.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/opentelemetry/otel_collector_datadog_exporter.md [hunk 1]
```diff
@@ -170,6 +170,12 @@ filelog:
     - `json_parser`: Parses JSON logs. By default, the filelog receiver converts each log line into a log record, which is the `body` of the logs' [data model][15]. Then, the `json_parser` converts the JSON body into attributes in the data model.
     - `trace_parser`: Extract the `trace_id` and `span_id` from the log to correlate logs and traces in Datadog. 
 
+#### Remap OTel service.name attribute to service in logs
+
+In certain Datadog Exporter versions, the `service` field of OTel logs is populated as [OTel semantic convention][25] `service.name`, and `service.name` is not in the default [service attributes][26] in Datadog log preprocessing.
```
Comment (maycmlee) [inline content/en/opentelemetry/otel_collector_datadog_exporter.md:175]:
```suggestion
In certain Datadog Exporter versions, the `service` field of OTel logs is populated as [OTel semantic convention][25] `service.name`. However, `service.name` is not one of the default [service attributes][26] in Datadog's log preprocessing.
```

File: content/en/opentelemetry/otel_collector_datadog_exporter.md [hunk 2]
```diff
@@ -170,6 +170,12 @@ filelog:
...
+
+To get the `service` field properly populated in logs, you can specify `service.name` to be the source of a log’s service by setting a [log service remapper processor][27].
```
Comment (maycmlee) [inline content/en/opentelemetry/otel_collector_datadog_exporter.md:177]:
```suggestion
To get the `service` field correctly populated in logs, you can specify `service.name` to be the source of a log's service by setting a [log service remapper processor][27].
```

File: content/en/opentelemetry/otel_collector_datadog_exporter.md [hunk 3]
```diff
@@ -170,6 +170,12 @@ filelog:
     - `json_parser`: Parses JSON logs. By default, the filelog receiver converts each log line into a log record, which is the `body` of the logs' [data model][15]. Then, the `json_parser` converts the JSON body into attributes in the data model.
     - `trace_parser`: Extract the `trace_id` and `span_id` from the log to correlate logs and traces in Datadog. 
 
+#### Remap OTel service.name attribute to service in logs
```
Comment (maycmlee) [inline content/en/opentelemetry/otel_collector_datadog_exporter.md:173]:
```suggestion
#### Remap OTel's `service.name` attribute to `service` for logs
```

**Review by maycmlee — COMMENTED** [review]
Left some feedback! Which Datadog Exporter versions is affected by this? It'd be helpful for users to know.

**Review by songy23 — COMMENTED** [review]
Thanks for the suggestions!

> Which Datadog Exporter versions is affected by this? It'd be helpful for users to know.

This is since v0.83.0 till the current version. We are not sure yet when the fix will be rolled out.

**Comment by songy23** [comment]
Do you think it's still useful to include the affected versions? For now I think we can only say "since v0.83.0"

File: content/en/opentelemetry/otel_collector_datadog_exporter.md [hunk 4]
```diff
@@ -170,6 +170,12 @@ filelog:
     - `json_parser`: Parses JSON logs. By default, the filelog receiver converts each log line into a log record, which is the `body` of the logs' [data model][15]. Then, the `json_parser` converts the JSON body into attributes in the data model.
     - `trace_parser`: Extract the `trace_id` and `span_id` from the log to correlate logs and traces in Datadog. 
 
+#### Remap OTel's `service.name` attribute to `service` for logs
+
+In certain Datadog Exporter versions, the `service` field of OTel logs is populated as [OTel semantic convention][25] `service.name`. However, `service.name` is not one of the default [service attributes][26] in Datadog's log preprocessing.
```
Comment (maycmlee) [inline content/en/opentelemetry/otel_collector_datadog_exporter.md:175]:
```suggestion
For Datadog Exporter versions 0.83.0 and later, the `service` field of OTel logs is populated as [OTel semantic convention][25] `service.name`. However, `service.name` is not one of the default [service attributes][26] in Datadog's log preprocessing.
```

**Review by maycmlee — COMMENTED** [review]
Hi, yes I do think it would be helpful for users to know which versions are affected. Once it has been fixed, we can update the info. This is my suggested update:

**Comment by songy23** [comment]
Thanks, that makes sense to me

File: content/en/opentelemetry/otel_collector_datadog_exporter.md [hunk 5]
```diff
@@ -170,6 +170,12 @@ filelog:
     - `json_parser`: Parses JSON logs. By default, the filelog receiver converts each log line into a log record, which is the `body` of the logs' [data model][15]. Then, the `json_parser` converts the JSON body into attributes in the data model.
     - `trace_parser`: Extract the `trace_id` and `span_id` from the log to correlate logs and traces in Datadog. 
 
+#### Remap OTel's `service.name` attribute to `service` for logs
+
+For Datadog Exporter versions 0.83.0 and later, the `service` field of OTel logs is populated as [OTel semantic convention][25] `service.name`. However, `service.name` is not one of the default [service attributes][26] in Datadog's log preprocessing.
+
+To get the `service` field correctly populated in logs, you can specify `service.name` to be the source of a log's service by setting a [log service remapper processor][27].
```
Comment (maycmlee) [inline content/en/opentelemetry/otel_collector_datadog_exporter.md:177]:
```suggestion
To get the `service` field correctly populated in your logs, you can specify `service.name` to be the source of a log's service by setting a [log service remapper processor][27].
```

**Review by maycmlee — APPROVED** [review]

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**